### ISSUE TITLE
[Bug]: TableToobarSearch rendering aria-label based on placeholder value, but i think should be based on labelText

### ISSUE THREAD

**Issue by David-Yuen** [carbon-design-system/carbon#18330]
### Package

@carbon/react

### Package version

1.72.0

### React version

18.3.1

### Description

I created a DataTable with TableToolbar components, including the Search component.
When configuring the `TableToolbarSearch` component, generally there are these 2 fields:
```
<TableToolbarSearch
  labelText="my label"
  placeholder="placeholder"
/>
```

And when it renders on the page, it generates something like this:
```
<div role="search" aria-label="..." ...>
  <label id="search-label-id" ..>
  <input ..>
  <button ..>
</div>
```

The problem is i see is, for the parent `div` with `role="search"` attribute, it creates an aria-label value based on the `placeholder` prop.. But instead, i think it should be based on `labelText` value. Doesn't that make more sense? 
The alternative is, the `div` could have `aria-labelledby` and reference the `id` value from the `<label>` element.. (example: `aria-labelledby="search-label-id"`)

When i render this component, generally `labelText` isnt really visible in this view (pretty much hidden). But the labelText seems like it should provide context of what this field is for.. so if aria-label was set to the same value as labelText, then in accessibility cases, the labelText value would be used appropriately. 
Compared to placeholder value, placeholder sometimes is a sample value like `1234` (example: Having an input field for `port` value, and the placeholder could be "1234").

### Reproduction/example

https://stackblitz.com/edit/github-nhj83gmf?file=src%2FApp.jsx

### Steps to reproduce

Generally, if you set `placeholder` value in the `TableToolbarSearch` component, it will get set as the `aria-label` value in the parent div. You will need to open the dev console and inspect the page elements to see. 

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by adamalston** [comment]
Is this issue a duplicate of https://github.com/carbon-design-system/carbon/issues/12360? If not, is this functionality working as designed per the comments in https://github.com/carbon-design-system/carbon/pull/13306? The alternative you're proposing is what was implemented before the changes in that pull request were introduced.
